crates/matrix-sdk-ui/tests/integration/timeline/subscribe.rs

```diff
@@ -173,9 +175,12 @@ async fn event_filter() {
     assert_let!(Some(VectorDiff::PushBack { value: second }) = timeline_stream.next().await);
     let second_event = second.as_event().unwrap();
     assert_eq!(second_event.event_id(), Some(second_event_id));
+    assert_eq!(second_event.read_receipts().len(), 1, "implicit read receipt");
 
-    // The implicit read receipt of Alice is updated.
-    assert_matches!(timeline_stream.next().await, Some(VectorDiff::Set { index: 1, .. }));
+    // The implicit read receipt of Alice is moving from Alice's message...
+    assert_let!(Some(VectorDiff::Set { index: 1, value: first }) = timeline_stream.next().await);
+    assert_eq!(first.as_event().unwrap().read_receipts().len(), 0, "no more implicit read receipt");
+    // … to Alice's edit. But since this item isn't visible, it's lost in the weeds!
```

Comment:
(interestingly, since the `event_filter` is set to `|_| true`, aka the previous default, this shows the buggy behavior, where a read receipt is attached to an invisible edit event. This is by design, so it's "fine" in this test.)